pkg/kubelet/kubelet.go

```diff
@@ -298,6 +304,9 @@ func NewMainKubelet(
 		resolverConfig:                 resolverConfig,
 		cpuCFSQuota:                    cpuCFSQuota,
 		daemonEndpoints:                daemonEndpoints,
+		authenticator:                  authenticator,
```

Comment:
You're saying these three things are logically grouped by an `AuthInterface`.  Why not group them in the kubelet in simple containing struct instead of exposing the methods raw on the kubelet?